sdk/ml/azure-ai-ml/azure/ai/ml/_schema/monitoring/monitor_definition.py

```diff
@@ -16,12 +16,13 @@
     CustomMonitoringSignalSchema,
 )
 from azure.ai.ml._schema.monitoring.alert_notification import AlertNotificationSchema
-from azure.ai.ml._schema.core.fields import NestedField, UnionField, ComputeField, StringTransformedEnum
+from azure.ai.ml._schema.core.fields import NestedField, UnionField, StringTransformedEnum
 from azure.ai.ml._schema.core.schema import PatchedSchemaMeta
+from azure.ai.ml._schema.spark_resource_configuration import SparkResourceConfigurationSchema
 
 
 class MonitorDefinitionSchema(metaclass=PatchedSchemaMeta):
-    compute = ComputeField()
+    compute = NestedField(SparkResourceConfigurationSchema)
```

Comment:
It's unfortunate we have to make a `compute` filed complex, which breaks the consistency with jobs.

Reply:
Agreed, but unfortunately amlcompute isn't supported. We should probably just rename this to something like `spark_configuration` for the public preview refresh.